tools/hdf_dump.py

```diff
@@ -63,6 +63,12 @@ def hdf_dump_from_dataset(dataset, hdf_dataset, parser_args):
     default_data_target_key = data_target_keys[0]
   print("Using target data key:", default_data_target_key)
 
+  if default_data_target_key == "data":
+    valid_hdf_target_key = "classes"  # Replace "data" which is reserved for input key in HDFDataset.
+    assert valid_hdf_target_key not in data_target_keys
+  else:
+    valid_hdf_target_key = default_data_target_key
```

Comment:
You need to introduce `hdf_data_key_map` here. It should be sth like:
```
hdf_data_key_map = {key: key for key in data_keys if key != default_data_input_key}
```
And then you can handle `"data"`. However, there should be extra checks for `"classes"`. E.g. like:
```
if "data" in hdf_data_key_map:
  # Replace "data" which is reserved for input key in HDFDataset.
  hdf_data_key_map["data"] = "classes"
  assert "classes" not in hdf_data_key_map
```

Reply:
Done.